### DISCUSSION TITLE
cannot get an ID token

### DISCUSSION THREAD

**Discussion by HosseinVenturies** [ercot/api-specs#60] [Q&A] [answered]
Hi there,

I am using the following Get request and only replacing the username and password with my personal one, 

https://ercotb2c.b2clogin.com/ercotb2c.onmicrosoft.com/B2C_1_PUBAPI-ROPC-FLOW/oauth2/v2.0/token?username=username@email.com&password=**********&grant_type=password&scope=openid+fec253ea-0d06-4272-a5e6-b478baeecd70+offline_access&client_id=fec253ea-0d06-4272-a5e6-b478baeecd70&response_type=id_token

and get an "invalid_request":
{
    "error": "invalid_request",
    "error_description": "AADB2C90226: The specified token exchange is only supported over HTTP POST.\r\nCorrelation ID: 5731dc2a-65c2-4885-b2dd-1d212c795c28\r\nTimestamp: 2024-11-04 22:01:03Z\r\n"
}

Can anyone help me with what I am missing?

Thanks!

**Comment by CaffeineLab** [comment, marked as answer]
The answer is in the response.  You need to POST the token request, not GET.

Check out the example by mackermann here:  [https://github.com/ercot/api-specs/discussions/34](https://github.com/ercot/api-specs/discussions/34)

**Comment by HosseinVenturies** [comment]
Fixed. Thanks!
